### ISSUE TITLE
[BUG] Intermittent test failures

### ISSUE THREAD

**Issue by da-woods** [cython/cython#5833]
### Describe your issue

There are two intermittent test failures on the CI that it would be nice not to have

- [x] `bytearray_coercion` on PyPy. This looks like `PyBytearray_AS_BYTES` either isn't returning a `\0` terminated string, or that the validity of the `char*` it returns isn't being maintained for long enough. We hold a real reference rather than a borrowed reference to the argument so its probably not that. I couldn't reproduce it locally (running about 10 times) with pypy3.10-v7.3.12-linux64, which was picked because it's the most up-to-date version already on my disk.

    Example of failing output

    ```
    File "/home/runner/work/cython/cython/TEST_TMP/5/run/c/bytearray_coercion/bytearray_coercion.pypy39-pp73-x86_64-linux-gnu.so", line ?, in bytearray_coercion.coerce_to_charptr
    Failed example:
        coerced == b or (coerced, b)
    Expected:
        True
    Got:
        (bytearray(b'abc\x1b\xf1\x7f'), bytearray(b'abc'))
   ```

    @mattip - any thoughts on this one
- [ ] Line profiling on Windows + Python 3.12. I think we're suspicious all forms of profiling on Python 3.12, but line profiling is possibly using the old mechanism to OK. It only looks to fail occasionally on Windows. Example of failing output:

    ```
     ======================================================================
    FAIL: disable_trace (line_trace)
    Doctest: line_trace.disable_trace
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "C:\hostedtoolcache\windows\Python\3.12.0\x64\Lib\doctest.py", line 2228, in runTest
        raise self.failureException(self.format_failure(new.getvalue()))
    AssertionError: Failed doctest test for line_trace.disable_trace
      File "D:\a\cython\cython\TEST_TMP\1\run\c\line_trace\line_trace.cp312-win_amd64.pyd", line unknown line number, in disable_trace

    ----------------------------------------------------------------------
    File "D:\a\cython\cython\TEST_TMP\1\run\c\line_trace\line_trace.cp312-win_amd64.pyd", line ?, in line_trace.disable_trace
    Failed example:
        disable_trace(py_add, 1, 2, with_sys=True)
    Expected:
        [('call', 0), ('line', 1)]
    Got:
        [('call', 0), ('call', 0)]
    ```
    Output seems reproducible (when it fails). I suspect we're OK  to just disable this test.

**Comment by mattip** [comment]
>  I couldn't reproduce it locally (running about 10 times) with pypy3.10-v7.3.12-linux64

Right, it seems to be a problem with the newly released 7.3.13 (as of about a month ago), and only on windows.

> I suspect we're OK to just disable this test

Yes please, maybe add it to #5832?

**Comment by da-woods** [comment]
> Right, it seems to be a problem with the newly released 7.3.13 (as of about a month ago), and only on windows.

Thanks. I don't think it's windows-only - it's appearing intermittently on our PyPy runs which are Ubuntu. I think I confused things here by having an intermittent CPython Windows failure in the same issue.

If it's fixed in a forthcoming release I think I'll just leave it rather than having to remember to re-enable the test.
